Tests/RudderStackAnalyticsTests/StorageModuleTests/BasicStorageTests.swift

```diff
@@ -0,0 +1,340 @@
+    private let sampleEventJson = """
+        {"messageId":"test-msg-123","type":"track","event":"Test Event","properties":{"test":true}}
+        """
+    var storage: BasicStorage
+    
+    init() {
+        storage = BasicStorage(writeKey: testWriteKey, storageMode: .memory)
+    }
+    
+    // MARK: - KeyValueStorage Tests
+    
+    @Test("given BasicStorage, when writing and reading string value, then value is persisted correctly")
+    func testKeyValueStringStorage() async {
+        let key = "test_string_key"
+        let value = "test_string_value"
+        
+        storage.write(value: value, key: key)
+        let retrievedValue: String? = storage.read(key: key)
+        
+        #expect(retrievedValue == value)
+    }
+    
+    @Test("given BasicStorage, when writing and reading integer value, then value is persisted correctly")
+    func testKeyValueIntegerStorage() async {
+        let key = "test_int_key"
+        let value = 42
+        
+        storage.write(value: value, key: key)
+        let retrievedValue: Int? = storage.read(key: key)
+        
+        #expect(retrievedValue == value)
+    }
+    
+    @Test("given BasicStorage, when writing and reading boolean value, then value is persisted correctly")
+    func testKeyValueBooleanStorage() async {
+        let key = "test_bool_key"
+        let value = true
+        
+        storage.write(value: value, key: key)
+        let retrievedValue: Bool? = storage.read(key: key)
+        
+        #expect(retrievedValue == value)
+    }
+    
+    @Test("given BasicStorage, when writing and reading codable object, then object is persisted correctly")
+    func testKeyValueCodableStorage() async {
+        struct TestObject: Codable, Equatable {
+            let id: String
+            let name: String
+            let count: Int
+        }
+        
+        let key = "test_object_key"
+        let value = TestObject(id: "123", name: "Test Object", count: 5)
+        
+        storage.write(value: value, key: key)
+        let retrievedValue: TestObject? = storage.read(key: key)
+        
+        #expect(retrievedValue == value)
+    }
+    
+    @Test("given BasicStorage, when reading non-existent key, then nil is returned")
+    func testKeyValueReadNonExistentKey() async {
+        let nonExistentKey = "non_existent_key"
+        let retrievedValue: String? = storage.read(key: nonExistentKey)
+        
+        #expect(retrievedValue == nil)
+    }
+    
+    @Test("given BasicStorage with stored value, when removing key, then value is deleted")
+    func testKeyValueRemove() async {
+        let key = "test_remove_key"
+        let value = "test_value_to_remove"
+        
+        storage.write(value: value, key: key)
+        var retrievedValue: String? = storage.read(key: key)
+        #expect(retrievedValue == value)
+        
+        storage.remove(key: key)
+        retrievedValue = storage.read(key: key)
+        #expect(retrievedValue == nil)
+    }
```

Comment:
We could write it like this:

```suggestion
    @Test("given value is stored in the storage, when key is removed, then value is deleted")
    func testKeyValueRemove() async {
        // This is the given portion - where we are storing the values
        let key = "test_remove_key"
        let value = "test_value_to_remove"
        storage.write(value: value, key: key)

        // This is the when portion, where we are removing the key
        storage.remove(key: key)

        // This is the then portion, for verifying our test
        retrievedValue = storage.read(key: key)
        #expect(retrievedValue == nil)
    }
```

Reply:
In your case, I see two issues:
1. Code is not properly strucutred i.e., `given.. when.. then..` is not aligned. Refer to my above code for structured code.
2. We unnecessarily verify something, even before performing `when` phase i.e., this line `#expect(retrievedValue == value)`.

Our aim should be to first write the `given` phase, then write the actual `when` phase and then at last write the `then` phase. We should try to avoid premature verification of anything.

Just a tip: Also, if there's no `given` phase, you may continue with the `when` and `then` phase.

Reply:
I commented the test case with steps.